programs/margin/src/state/account/positions.rs

```diff
@@ -267,13 +288,24 @@ impl AccountPosition {
     }
 
     pub fn may_be_registered_or_closed(&self, approvals: &[Approver]) -> bool {
-        approvals.contains(&Approver::MarginAccountAuthority)
-            && match self.kind() {
-                PositionKind::NoValue | PositionKind::Deposit => true,
-                PositionKind::Claim | PositionKind::AdapterCollateral => {
-                    approvals.contains(&Approver::Adapter(self.adapter))
-                }
+        let mut authority_approved = false;
+        let mut adapter_approved = false;
+
+        for approval in approvals {
+            match approval {
+                Approver::MarginAccountAuthority => authority_approved = true,
+                Approver::Adapter(_) => adapter_approved = true,
```

Comment:
Seems that the main thing you changed here was that you removed the adapter restriction. You're allowing any adapter to register a position regardless of which adapter is in charge of that token. So for example the bonds program can register a loan in pools. I don't think we should allow that kind of thing.
```suggestion
                Approver::Adapter(approving_adapter) => adapter_approved = approving_adapter == self.adapter,
```